tests/Feature/Http/Controllers/AnnounceControllerTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Tests\Feature\Http\Controllers;
+
+use App\Helpers\Bencode;
+use App\Models\Torrent;
+use App\Models\User;
+use GroupsTableSeeder;
+use Tests\TestCase;
+
+/**
+ * @see \App\Http\Controllers\AnnounceController
+ */
+class AnnounceControllerTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function announce_returns_an_ok_response()
+    {
+        $this->seed(GroupsTableSeeder::class);
+
+        $user = factory(User::class)->create([
+            'can_download' => true,
+        ]);
+
+        factory(Torrent::class)->create([
+            'info_hash' => bin2hex(md5('foo')),
+            'status'    => 1, // Approved
+        ]);
+
+        $response = $this->get(route('announce', [
+            'passkey'    => $user->passkey,
+            'info_hash'  => md5('foo'),
```

Comment:
Possibly better to make like 27 from `factory(Torrent::class)->create([.....` to `$torrent = factory(Torrent::class)->create([......` and then reference `'info_hash'  => $torrent->info_hash,` here instead?

Reply:
Indeed; I've made this change. (The flow is a little different than as suggested because the info hash is `bin2hex()`'d in the database, but not when the client sends it.)